packages/nimble-components/src/table/components/group-row/index.ts

```diff
@@ -0,0 +1,88 @@
+import { attr, html, observable, ViewTemplate } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import type { TableColumn } from '../../../table-column/base';
+import { styles } from './styles';
+import { template } from './template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-group-row': TableGroupRow;
+    }
+}
+
+/**
+ * A styled cell that is used within the nimble-table-row.
+ * @internal
+ */
+export class TableGroupRow extends FoundationElement {
+    @observable
+    public groupRowValue?: unknown;
+
+    @observable
+    public nestingLevel = 0;
+
+    @observable
+    public leafItemCount?: number;
+
+    @observable
+    public groupColumn?: TableColumn;
+
+    @attr({ mode: 'boolean' })
+    public expanded = false;
+
+    /**
+     * @internal
+     */
+    @observable
+    public groupHeaderViewTemplate?: ViewTemplate<TableGroupRow>;
+
+    /**
+     * @internal
+     */
+    public readonly expandIcon!: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public animationClass = '';
+
+    public onGroupExpandToggle(): void {
+        this.$emit('group-expand-toggle');
+        this.animationClass = 'animating';
+        this.expandIcon.addEventListener(
+            'transitionend',
+            this.removeAnimatingClass
```

Comment:
This event handling to add/remove a class representing the expand/collapse button animation is to avoid a visual bug that occurs during expand/collapse due to the visual re-use of rows. This will basically guarantee that the only button that animates will be the one with the class present on it.